### PR TITLE
New: Blog post announcing v7.0.0 features

### PR DESCRIPTION
**Pull request by nzakas**
This is a blog post previewing the significant features and changes coming in v7.0.0. I did my best to explain the rationale behind some of the bigger changes, but do let me know if I didn't get it right.

Note that I only included what I thought to be significant changes, so please feel free to let me know if I missed anything. I didn't think changes to rules were important enough to merit inclusion (those will be better explained in the release notes).

### CODE REVIEW

**Comment by platinumazure** [comment]
This post mentions releases happening every two weeks, but (as of when I was on the team) we were releasing every four weeks. Is ESLint releasing every two weeks again? (Kai was hired on as dedicated maintainer after the decision to go to four weeks, so the landscape definitely changed after we made that decision.)

**Comment by kaicataldo** [comment]
@platinumazure We're not releasing every two weeks, though I'm happy to be the point person to do so.

File: _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+Support for home directory config files will be removed in v8.0.0. You can read more in the [RFC](https://github.com/eslint/rfcs/pull/32).
+
+### Updating the base path when using `--config` or `--ignore-path`
+
+In versions prior to v7.0.0, ESLint would resolve references inside of config files or ignore files relative to the file location passed to `--config` or `--ignore-path`. This didn't make much sense and was a source of confusion and unexpected behaviors for users. In v7.0.0, config files and ignore files passed using `--config` or `--ignore-path` use the current working directory as their base path. You can read more in the [RFC](https://github.com/eslint/rfcs/pull/37).
+
+
+### File extensions in config files
+
+ESLint v7.0.0 now reads the glob patterns in the `overrides` section of your config to determine while files to lint. Prior to v7.0.0, you would need to provide the `--ext` command line flag to specify the additional files to lint. You can read more in the [RFC](https://github.com/eslint/rfcs/pull/20).
```
Comment (kaicataldo) [inline _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md]:
```suggestion
ESLint v7.0.0 now reads the glob patterns in the `overrides` section of your config to determine which files to lint. Prior to v7.0.0, you would need to provide the `--ext` command line flag to specify the additional files to lint. You can read more in the [RFC](https://github.com/eslint/rfcs/pull/20).
```

File: _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+### Updated default ignore patterns
+
+The default ignore patterns for ESLint will be changed to:
```
Comment (kaicataldo) [inline _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md]: Can we make it clear that these are additive changes? Maye something like:
```suggestion
In addition to the current defaults, the default ignore patterns for ESLint will be changed to:
```

**Comment by kaicataldo** [comment]
Thanks for writing this up! Two small comments, but otherwise LGTM.

**Comment by nzakas** [comment]
Thanks. Apparently I can’t merge changes from my phone so I’ll have to do that later.

I didn’t realize we had changed to every four weeks for releases. Is there any background to that decision you can point me to? FWIW, the readme still says every two weeks.

File: _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+```js
+/* eslint-disable no-new -- this class has a side-effect in the constructor. */
+```
+
+You can read more in the [RFC](https://github.com/eslint/rfcs/pull/33).
+
+### Deprecating `~/.eslintrc`
+
+Config files in your home directory are deprecated in v7.0.0. This feature has long been a source of confusion for ESLint users, especially because ESLint only looks for a config file in the home directory if no other config file is found. The configuration system in ESLint has changed a lot since home directory config files were introduced, and with the additional flexibility we feel there are plenty of other options for configuring ESLint that are clearer than the home directory config file.
```
Comment (mysticatea) [inline _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md]: This deprecation happened in 6.7.0. We add runtime warnings in 7.0.0 for a few breaking cases on removal.

**Comment by ilyavolodin** [comment]
Switch to once-per-4 weeks release happened as part of TSC discussion a few month ago. The reasons for it were that we couldn't find people to cover releases every 2 week. Suggestions were to either switch back to requiring one person per release, or to switch to release every 4 weeks. We missed a bunch of scheduled releases in December because only one person was available to cover release.

File: _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md (hunk 3 above)
Comment (nzakas) [reply _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md]: Ooh thanks, I'll update that.

**Comment by nzakas** [comment]
@ilyavolodin thanks for the details.

**Comment by nzakas** [comment]
Okay, I've addressed all of the comments. Please take a look and let me know if anything else should be updated before publishing. (I'd like to publish today, so please take a look as soon as you can.)

**Comment by kaicataldo** [comment]
Thanks for writing this up!

File: _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md [hunk 4]
```diff
@@ -58,16 +58,19 @@ Directive comment descriptions allow you to leave a note to yourself and other d
...
 
 Support for home directory config files will be removed in v8.0.0. You can read more in the [RFC](https://github.com/eslint/rfcs/pull/32).
 
 ### Updating the base path when using `--config` or `--ignore-path`
 
 In versions prior to v7.0.0, ESLint would resolve references inside of config files or ignore files relative to the file location passed to `--config` or `--ignore-path`. This didn't make much sense and was a source of confusion and unexpected behaviors for users. In v7.0.0, config files and ignore files passed using `--config` or `--ignore-path` use the current working directory as their base path. You can read more in the [RFC](https://github.com/eslint/rfcs/pull/37).
 
+### Plugins loaded from config file directory
+
+ESLint has long had a problem loading plugins from shared configs due to the way plugin loading was handled. This problem required using peer dependencies and a post-install script to allow shared configs to reference plugins. In v7.0.0, plugins will be loaded relative to the configs that reference them, allowing shared configs to include plugins as dependencies directly. You can read more in the [RFC](https://github.com/eslint/rfcs/pull/47).
```
Comment (mysticatea) [inline _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md]: > allowing shared configs to include plugins as dependencies directly.

No.
RFC47 doesn't intend to allow shared configs to include plugins as dependencies directly. See detail: https://github.com/eslint/rfcs/blob/master/designs/2019-plugin-loading-improvement/README.md#the-plugin-uniqueness-with-shareable-configs

Comment (nzakas) [reply _posts/2020-02-20-whats-coming-in-eslint-7.0.0.md]: Oh thanks. That RFC was a bit hard to understand for me, so I missed this part.

**Pull request merged**